js/stores/windowStore.js

```diff
@@ -206,8 +209,12 @@ const doAction = (action) => {
       break
     case WindowConstants.WINDOW_NEW_FRAME:
       let nextKey = incrementNextKey()
+      let nextPartitionNumber = currentPartitionNumber
+      if (action.frameOpts.isPartitioned) {
+        nextPartitionNumber = incrementPartitionNumber()
+      }
```

Comment:
This is actually a bug. The correct functionality for new tab is to always open with the default session 0.  
Tabs and windows opened from a partitioned tab inherit that tab though.
Will fix, nice find.